src/core/include/openvino/op/constant.hpp

```diff
@@ -290,48 +290,46 @@ class OPENVINO_API Constant : public Op {
     ///
     /// \tparam T             Type to which data vector's entries will be cast.
     /// \param  num_elements  (Optional) Number of elements to cast. In default case returns all elements
+    /// \param  validate_data (Optional) Enable validation data on casting. Default false, no validation.
     /// \return    Constant's data vector.
     template <typename T>
-    std::vector<T> cast_vector(int64_t num_elements = -1) const {
+    std::vector<T> cast_vector(int64_t num_elements = -1, bool validate_data = false) const {
```

Comment:
May I ask not to introduce this parameter -- do you really set it to `true` somewhere? I think nobody needs it -- when it is required we will add it.

Reply:
I just want to eliminate additional burden in the functionality that nobody needs.

Reply:
Yes, I can remove as it looks like not required  from our CI tests perspective.
Just need approval to do this as there change in default behavior.

Remove this validation will improve speed and also reduce binary size as there a lot of code generate for many types.

The data validation can be moved as user responsibility. The user can still get Constant data as bytes and interpret it according Constant's precision then check values.
In OV if it will be required the test or dev API helper function can be added to do this validation.

Reply:
I found out that the check was added in https://github.com/openvinotoolkit/openvino/pull/18369
Tickets were CVS-97764 (compilation warning) and CVS-113304 (fuzz tests). I don't see how this is customer-oriented. Making cast_vector faster is super customer-oriented -- our Python bindings are sometimes using cast_vector to make a copy of constant data -- so, making this code faster is more important than fuzz tests and compilation warnings.